src/libraries/Common/src/Roslyn/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -48,12 +74,16 @@ internal static partial class SyntaxValueProviderExtensions
         // changed. CreateSyntaxProvider will have to rerun all incremental nodes since it passes along the
         // SemanticModel, and that model is updated whenever any tree changes (since it is tied to the compilation).
         var syntaxTreesProvider = context.CompilationProvider
-            .SelectMany(static (c, _) => c.SyntaxTrees)
+            .SelectMany((compilation, cancellationToken) => compilation.SyntaxTrees
+                .Select(tree => GetTreeInfo(tree, syntaxHelper, cancellationToken))
+                .Where(info => info.ContainsGlobalAliases || info.ContainsAttributeList))
```

Comment:
this `Where` filters the entire set of trees (normally thousands) to the much smaller set of interest.  ideally in most projects this will be orders of magnitude less trees (or 0 trees in some projects).  After all, if you're not using attributes, why do we even need to analyze anything :)